examples/lambda/README.md

```diff
@@ -0,0 +1,279 @@
+- `consul-lambda-registrator` is configured to manage Lambda functions within the mesh:
+  - AWS EventBridge is configured to invoke `consul-lambda-registrator` periodically to perform a full synchronization of AWS Lambda state with Consul.
+  - AWS EventBridge is configured to monitor Lambda events from CloudTrail to synchronize the state of individual AWS Lambda functions with Consul, when their state changes.
+  - `consul-lambda-registrator` stores the required mTLS material in AWS Systems Manager Parameter Store.
+- `lambda-app-1` is a Lambda function that accepts incoming requests from the `ecs-app-1` service.
+- `lambda-app-2` is a Lambda function that makes outgoing requests to the `ecs-app-2` service.
+- `consul-lambda-extension` is configured as a Lambda layer. The extension uses the mTLS artifacts from Parameter Store to proxy requests from `lambda-app-2` to the services in the mesh through the mesh gateway.
+
+_Note: The deployment configuration in this example is for demonstration purposes only and should not be considered best practice for production deployments._
+
+_Note: Some of the steps in this example will create AWS resources that may result in charges to your AWS account._
+
+## Prerequisites
+
+In order to run this example you will need the following tools installed:
+- Go
+- Terraform
+- AWS CLI
+- Docker
+- `curl`
+- `jq`
+
+## Set the version
+
+The subsequent steps in the example require a version identifier.
+To make the steps repeatable across releases, the target version is set in an environment variable.
+
+```shell
+export VERSION=<X.Y.Z>
+```
+
+Replace `<X.Y.Z>` with the target version of this repository, for example:
+
+```shell
+export VERSION=0.1.0
+```
+
+## Get the example code
+
+```shell
+git clone https://github.com/hashicorp/terraform-aws-consul-lambda
+cd terraform-aws-consul-lambda/examples/lambda
+git checkout v${VERSION}
+```
+
+## Set your AWS account ID and region
+
+Subsequent steps require knowledge of your AWS account ID and the AWS region that you want to deploy the example resources to.
+Export these values to environment variables using the commands below.
+Replace `<account_id>` and `<region>` with your AWS account ID and region, respectively.
+
+```shell
+export AWS_ACCOUNT_ID=<account_id>
+export AWS_REGION=<region>
+```
+
+## Publish `consul-lambda-registrator`
+
+In this section you will pull the `consul-lambda-registrator` image from the AWS Public ECR Gallery and publish it to a private ECR repository using `docker`. This is required because AWS Lambda functions must use images from a private ECR repository. They are not able to use images from the Public ECR Gallery.
+
+### Pull `consul-lambda-registrator`
+
+Use the following command to pull the `consul-lambda-registrator` from the AWS Public ECR to your local machine.
+
+```shell
+docker pull public.ecr.aws/hashicorp/consul-lambda-registrator:${VERSION}
```

Comment:
This doesn't work yet because the latest version of `consul-lambda-registrator` hasn't been deployed to ECR. During testing I pulled from `hashicorpdev/consul-lambda-registrator:<commit>`